cmake/HPX_SetupBoost.cmake

```diff
@@ -72,6 +72,8 @@ endif()
 list(REMOVE_DUPLICATES __boost_libraries)
 
 set(Boost_NO_BOOST_CMAKE ON) # disable the search for boost-cmake
+# solve set_property optimised pb with boost, no need when using SYSTEM option
+set(Boost_NO_SYSTEM_PATHS ON)
```

Comment:
Why does this solve the problem? And won't this keep CMake from finding a system-installed Boost automatically?

Reply:
It's just several tests to solve the travisci problem, I will not keep it that way ! I tried multiple things including `$<TARGET_PROPERTY:target,LINK_LIBRARIES>` and still not working, would you have any other ideas ?